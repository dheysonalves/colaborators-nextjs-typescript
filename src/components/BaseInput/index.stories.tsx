import React from 'react'

import { ComponentMeta, ComponentStory } from '@storybook/react'
import BaseInput from '.'

export default {
	title: 'Inputs/BaseInput',
	component: BaseInput,
	args: {
		label: 'Pesquise por',
		placeholderText: 'Pesquise por nome ou cpf'
	}
} as ComponentMeta<typeof BaseInput>

export const Default: ComponentStory<typeof BaseInput> = (args) => {
	return <BaseInput {...args} />
}
