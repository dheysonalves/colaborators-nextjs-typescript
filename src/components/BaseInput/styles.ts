import styled from '@emotion/styled'
import { Container } from '../utils/Flex/flex.styles'

export const Wrapper = styled.div`
	position: relative;
	padding: 8px 14px;
	border-radius: 8px;
	border: 2px solid #cad6d1;
	background-color: #fff;

	label {
		position: absolute;
		top: -11px;
		left: 20px;
		background-color: white;
		padding: 0px 3px;
	}

	input {
		border: none;
		-webkit-appearance: none;
		outline: none;
		width: 100%;
	}
`
export const IconWrapper = styled(Container)`
	margin-top: 5px;
	align-items: center;
`

export const Label = styled.label`
	color: #a3b8b0;
	font-size: 14px;
	font-weight: 500;
`

export const Input = styled.input`
	color: #587169;
	font-weight: 500;
	font-size: 14px;
	line-height: 24px;
	margin-left: 8px;
`
