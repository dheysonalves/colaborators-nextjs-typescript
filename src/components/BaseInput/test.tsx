import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import BaseInput from '.'

const setup = () => {
	const utils = render(
		<BaseInput
			placeholderText="Pesquise por"
			label="Pesquise por nome ou cpf"
		/>,
	)
	const input = utils.getByLabelText('base-search-input') as HTMLInputElement
	return {
		input,
		...utils,
	}
}

describe('<BaseInput />', () => {
	it('it should type Colaboradores in text-input', () => {
		const { input, container } = setup()

		fireEvent.change(input, { target: { value: 'Colaboradores' } })
		expect(input.value).toBe('Colaboradores')

		expect(container.firstChild).toMatchSnapshot()
	})

	it('it should allow Colaboradores to be deleted', () => {
		const { input, container } = setup()

		fireEvent.change(input, { target: { value: 'Colaboradores' } })
		expect(input.value).toBe('Colaboradores')
		fireEvent.change(input, { target: { value: '' } })
		expect(input.value).toBe('')

		expect(container.firstChild).toMatchSnapshot()
	})
})
