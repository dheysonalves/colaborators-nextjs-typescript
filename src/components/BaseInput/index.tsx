import React from 'react'
import * as S from './styles'
import { FiSearch } from 'react-icons/fi'

interface BaseInputProps {
	label: string
	placeholderText: string
}

const BaseInput = ({ label, placeholderText }: BaseInputProps) => {
	const [text, updateText] = React.useState('')

	const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		e.preventDefault()
		updateText(e.target.value)
	}

	return (
		<S.Wrapper>
			<S.Label htmlFor="input-search">{label}</S.Label>
			<S.IconWrapper>
				<FiSearch size={24} color="#587169" />
				<S.Input
					type="text"
					name="input-search"
					id="input-search"
					aria-label="base-search-input"
					placeholder={placeholderText}
					value={text}
					onChange={(e) => handleChange(e)}
				/>
			</S.IconWrapper>
		</S.Wrapper>
	)
}

export default BaseInput
