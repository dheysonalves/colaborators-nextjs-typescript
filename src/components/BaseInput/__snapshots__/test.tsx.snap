// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> it should allow Colaboradores to be deleted 1`] = `
<div
  class="css-zb6yxz"
>
  <label
    class="css-au0td9"
    for="input-search"
  >
    Pesquise por nome ou cpf
  </label>
  <div
    class="css-1e3v0jf"
  >
    <svg
      color="#587169"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(88, 113, 105);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
    </svg>
    <input
      aria-label="base-search-input"
      class="css-1m5m2pc"
      id="input-search"
      name="input-search"
      placeholder="Pesquise por"
      type="text"
      value=""
    />
  </div>
</div>
`

exports[`<BaseInput /> it should type Colaboradores in text-input 1`] = `
<div
  class="css-zb6yxz"
>
  <label
    class="css-au0td9"
    for="input-search"
  >
    Pesquise por nome ou cpf
  </label>
  <div
    class="css-1e3v0jf"
  >
    <svg
      color="#587169"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(88, 113, 105);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
    </svg>
    <input
      aria-label="base-search-input"
      class="css-1m5m2pc"
      id="input-search"
      name="input-search"
      placeholder="Pesquise por"
      type="text"
      value="Colaboradores"
    />
  </div>
</div>
`
